### PR TITLE
Add --include-asn1 switch to pyasn1gen.py

### PR DESCRIPTION
**Pull request by kimgr**
This embeds the ASN.1 source into the generated Python code.

Since a Python file can contain multiple ASN.1 modules, the ASN1_SOURCES
constant is always a dict, and sources are keyed by ASN.1 module name.

### CODE REVIEW

File: asn1ate/support/pygen.py [hunk 1]
```diff
@@ -105,3 +105,22 @@ def __init__(self, indent_size=4):
...
+
+    \"\"\"
+    Hello
+    \"\"\"
+
+    Inline quote chars are escaped.
+    """
+    lines = [
+        '"""',
+        content.replace('"', '\\"'),
```
Comment (rtweeks) [inline asn1ate/support/pygen.py]: It might be good to also replace `\` with a double backslash, especially if a `\` can occur in the ASN.1.

Comment (kimgr) [reply asn1ate/support/pygen.py]: I completely lost track of this; spring is usually my most busy time of the year. But now, a year later, I had some spare time and rediscovered the branch. :)

You're right -- we want to escape both `\` and `"""`, but `\"` itself is not necessary.

I found a number of bugs in the Python code-generation for ASN.1 with exotic escape sequences, but this source embedding should work OK now.

**Comment by kimgr** [comment]
I pushed some fixups and a test case (that's not strictly used for `--include-asn1`, but handy nonetheless).

But I just realized that we're not including the source text -- only a regeneration of it from asn1ate's point of view. That is, `str(module)` regenerates a source document back from the semantic model. I don't know if that's a problem or not, but it might be an advantage, to be able to see how asn1ate sees the document. Splitting the source text into distinct modules seems like a bigger job at the moment.

**Comment by rtweeks** [comment]
I think it's fine to use the regenerated version, so long as it is
semantically the same. The real point is to allow the python code to share
the format of the data it understands in ASN.1 for language independence --
i.e. so additional tools can be written in other languages to use the same
binary representation (especially in DER).

This is critical for my application (of ASN.1, actually a tool/library),
because I use ASN.1 DER to express JSON-type data in a reproducible binary
format feeding a SHA256 hash computation, the output of which is used as a
table key in persistent data. If tools written in other languages want to
compute the key natively from the JSON data to do lookups in that table,
they need the ASN.1 to build their own binary format encoder. As an
example, I could imagine another developer wanting to write a library in
Haskell similar to the one I've written in Python; with my Python tool able
to spit out the ASN.1, that developer doesn't have to dig in my repository
for the ASN.1 (more important if it is otherwise proprietary), and this
feature allows me to accomplish that without the gyrations necessary to
include and read a non-Python file in my package.

**Comment by kimgr** [comment]
Thanks for the background and confirmation that this is good enough. Merging now.

**Pull request merged**